tests/components/here_travel_time/test_sensor.py

```diff
@@ -0,0 +1,721 @@
+            {
+                "latitude": float(TRUCK_ORIGIN.split(",")[0]),
+                "longitude": float(TRUCK_ORIGIN.split(",")[1])
+            }
+        )
+        hass.states.async_set(
+            'person.destination',
+            "unknown",
+            {
+                "latitude": float(TRUCK_DESTINATION.split(",")[0]),
+                "longitude": float(TRUCK_DESTINATION.split(",")[1])
+            }
+        )
+
+        config = {DOMAIN: {
+            'platform': PLATFORM,
+            'name': 'test',
+            'origin': "person.origin",
+            'destination': "person.destination",
+            'app_id': APP_ID,
+            'app_code': APP_CODE,
+            'mode': TRAVEL_MODE_TRUCK
+        }}
+        assert await async_setup_component(hass, DOMAIN, config)
+
+        sensor = hass.states.get('sensor.test')
+        _assert_truck_sensor(sensor)
+
+        # Test that update works more than once
+        async_fire_time_changed(hass, utcnow + SCAN_INTERVAL)
+        await hass.async_block_till_done()
+
+        sensor = hass.states.get('sensor.test')
+        _assert_truck_sensor(sensor)
+
+
+async def test_location_device_tracker(hass, requests_mock_truck_response):
+    """Test that origin/destination supplied by a device_tracker works."""
+    utcnow = dt_util.utcnow()
+    # Patching 'utcnow' to gain more control over the timed update.
+    with patch('homeassistant.util.dt.utcnow', return_value=utcnow):
+        hass.states.async_set(
+            'device_tracker.origin',
+            "unknown",
+            {
+                "latitude": float(TRUCK_ORIGIN.split(",")[0]),
+                "longitude": float(TRUCK_ORIGIN.split(",")[1])
+            }
+        )
+        hass.states.async_set(
+            'device_tracker.destination',
+            "unknown",
+            {
+                "latitude": float(TRUCK_DESTINATION.split(",")[0]),
+                "longitude": float(TRUCK_DESTINATION.split(",")[1])
+            }
+        )
+
+        config = {DOMAIN: {
+            'platform': PLATFORM,
+            'name': 'test',
+            'origin': "device_tracker.origin",
+            'destination': "device_tracker.destination",
+            'app_id': APP_ID,
+            'app_code': APP_CODE,
+            'mode': TRAVEL_MODE_TRUCK
+        }}
+        assert await async_setup_component(hass, DOMAIN, config)
+
+        sensor = hass.states.get('sensor.test')
+        _assert_truck_sensor(sensor)
+
+        # Test that update works more than once
+        async_fire_time_changed(hass, utcnow + SCAN_INTERVAL)
+        await hass.async_block_till_done()
+
+        sensor = hass.states.get('sensor.test')
+        _assert_truck_sensor(sensor)
+
+
+async def test_location_device_tracker_added_after_update(
+        hass, requests_mock_truck_response):
+    """Test that device_tracker added after first update works."""
+    with patch(
+        'homeassistant.components.here_travel_time.sensor._LOGGER.error'
```

Comment:
Don't patch the logger. We can use `caplog` fixture to check captured logging messages.

http://doc.pytest.org/en/latest/logging.html#caplog-fixture

Reply:
Awesome! Implemented it.